### DISCUSSION TITLE
Elm Backend

### DISCUSSION THREAD

**Discussion by KindsonTheGenius** [finos/morphir#188] [Proposal]
Seems this is the right time to consider creating an Elm backend  for at least three reasons:

- The current TypeBuilder modifies the Morphir IR thereby unsyncing it with the original Model. An Elm backend would allow us to generate the model for the modified IR (or update the original one).
- With an Elm backend, we could have custom codec generators for Elm custom types.
- Finally, sometime we may need to source-control the Morphir IR and Elm backend would be needed

What are our thoughts on this?

**Comment by DamianReeves** [comment]
We are also considering an Elm backend in `morphir-scala`

**Reply by AttilaMihaly** [reply]
That's exciting. Let's discuss on the community call in more detail.
